Frameworks/QuartzCore/CALayer.mm

```diff
@@ -325,60 +287,45 @@ static void DiscardLayerContents(CALayer* layer) {
     if (savedContext) {
         CGContextRelease(savedContext);
     }
-    [maskLayer release];
-    maskLayer = nil;
-
-    GetCACompositor()->ReleaseNode(_presentationNode);
-    _presentationNode = NULL;
 }
 
-class LockingBufferInterface : public DisplayTextureLocking {
-public:
-    void* LockWritableBitmapTexture(DisplayTexture* tex, int* stride) {
-        return GetCACompositor()->LockWritableBitmapTexture(tex, stride);
-    }
-    void UnlockWritableBitmapTexture(DisplayTexture* tex) {
-        GetCACompositor()->UnlockWritableBitmapTexture(tex);
-    }
-
-    void RetainDisplayTexture(DisplayTexture* tex) {
-        GetCACompositor()->RetainDisplayTexture(tex);
-    }
-
-    void ReleaseDisplayTexture(DisplayTexture* tex) {
-        GetCACompositor()->ReleaseDisplayTexture(tex);
-    }
-};
-
-static LockingBufferInterface _globallockingBufferInterface;
-
 CGContextRef CreateLayerContentsBitmapContext32(int width, int height) {
-    DisplayTexture* tex = NULL;
-
     if ([NSThread isMainThread]) {
-        tex = GetCACompositor()->CreateWritableBitmapTexture32(width, height);
-    }
-
-    CGContextRef ret = _CGBitmapContextCreateWithTexture(width, height, tex, &_globallockingBufferInterface);
-
-    if (tex) {
-        _globallockingBufferInterface.ReleaseDisplayTexture(tex);
+        std::shared_ptr<IDisplayTexture> texture = GetCACompositor()->CreateDisplayTexture(width, height);
+        return _CGBitmapContextCreateWithTexture(width, height, texture);
     }
 
-    return ret;
+    return nil;
 }
 
 @implementation CALayer
+
 /**
  @Status Interoperable
 */
 - (instancetype)init {
-    assert(priv == NULL);
-    priv = new CAPrivateInfo(self);
+    return [self _initWithXamlElement:nil];
+}
 
+/**
+ Microsoft Extension
+ All CALayers are ultimately backed by a Xaml FrameworkElement.  Passing nil here will
```

Comment:
that's a good idea; i'll follow up on it.

---
In reply to: [86620194](https://github.com/Microsoft/WinObjC/pull/1318#discussion_r86620194) [](ancestors = 86620194)